core/vibes/soul/sections/sign-in-section/sign-in-form.tsx

```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import { getFormProps, getInputProps, SubmissionResult, useForm } from '@conform-to/react';
+import { getZodConstraint, parseWithZod } from '@conform-to/zod';
+import { useActionState, useEffect } from 'react';
+import { useFormStatus } from 'react-dom';
+
+import { Input } from '@/vibes/soul/form/input';
+import { Button } from '@/vibes/soul/primitives/button';
+
+import { schema } from './schema';
+
+type Action<State, Payload> = (state: Awaited<State>, payload: Payload) => State | Promise<State>;
+
+export type SignInAction = Action<SubmissionResult | null, FormData>;
+
+interface Props {
+  action: SignInAction;
+  emailLabel?: string;
+  passwordLabel?: string;
+  submitLabel?: string;
+}
+
+export function SignInForm({
+  action,
+  emailLabel = 'Email',
+  passwordLabel = 'Password',
+  submitLabel = 'Sign in',
+}: Props) {
+  const [lastResult, formAction] = useActionState(action, null);
+  const [form, fields] = useForm({
+    defaultValue: { email: '', password: '' },
+    constraint: getZodConstraint(schema),
+    shouldValidate: 'onBlur',
+    shouldRevalidate: 'onInput',
+    onValidate({ formData }) {
+      return parseWithZod(formData, { schema });
+    },
+  });
+
+  useEffect(() => {
+    if (lastResult?.error) console.log(lastResult.error);
+  }, [lastResult]);
```

Comment:
I think that for a form it makes more sense to show the errors below the submit button instead of using an alert